packages/firebase_auth/lib/firebase_auth.dart

```diff
@@ -165,15 +165,19 @@ typedef void PhoneCodeSent(String verificationId, [int forceResendingToken]);
 typedef void PhoneCodeAutoRetrievalTimeout(String verificationId);
 
 class FirebaseAuth {
-  FirebaseAuth({FirebaseApp app})
-      : app = app != null ? app : FirebaseApp.instance {
+  FirebaseAuth._(this.app) {
```

Comment:
This is also a change to the PR: https://github.com/flutter/plugins/pull/781 and won't cause a breaking change.